src/App.tsx

```diff
@@ -1,14 +1,61 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
+import './App.scss';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './services/getTodos';
+import { Todo } from './types/Todo';
+import { FilterSelect } from './types/FilterSelect';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState<string>('');
+  const [filterSelect, setFilterSelect] = useState<FilterSelect>(FilterSelect.All);
+
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [selectTodo, setSelectTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    getTodos()
+      .then((data) => {
+        setTodos(data);
+      })
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      .catch((_error) => {
+        setTodos([]);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }, []);
+
+  const getFilteredTodos = () => {
+    let newTodos = [...todos];
+
+    newTodos = newTodos.filter((todo) => {
+      switch (filterSelect) {
+        case FilterSelect.Active:
+          return !todo.completed;
+
+        case FilterSelect.Completed:
+          return todo.completed;
+
+        default:
+          return true;
+      }
+    });
+
+    return newTodos.filter(({ title }) => title.trim().toLowerCase()
+      .includes(filter.trim().toLocaleLowerCase()));
+  };
```

Comment:
I think will be more correctly to move this function outside component and use useMemo hook